src/DataMigration/DataMigration.Autorest/custom/Cmdlets/New-AzDataMigrationLoginsMigration.ps1

```diff
@@ -78,77 +78,179 @@ function New-AzDataMigrationLoginsMigration
             #Defining Default Output Path
             $DefaultOutputFolder = Get-DefaultLoginMigrationsOutputFolder
 
-            #Defining Base and Exe paths
-            $BaseFolder = Join-Path -Path $DefaultOutputFolder -ChildPath Downloads;
-            $ExePath = Join-Path -Path $BaseFolder -ChildPath Logins.Console.csproj\Logins.Console.exe;
+            #Defining Downloads folder
+            $DownloadsFolder = Join-Path -Path $DefaultOutputFolder -ChildPath Downloads;
 
-            #Checking if BaseFolder Path is valid or not
-            if(-Not (Test-Path $BaseFolder))
+            #Checking if DownloadsFolder Path is valid or not
+            if(-Not (Test-Path $DownloadsFolder))
             {
-                $null = New-Item -Path $BaseFolder -ItemType "directory"
+                $null = New-Item -Path $DownloadsFolder -ItemType "directory"
+            }
+            else
+            {
+                #Remove old Login console app files
+                #Old console app download address
+                $ZipDestination = Join-Path -Path $DownloadsFolder -ChildPath "LoginsMigration.zip";
+
+                # Remove old zip file
+                if(Test-Path $ZipDestination)
+                {
+                    Remove-Item -Path $ZipDestination;
+                }
+
+                # Remove existing folder and contents
+                $ConsoleAppDestination = Join-Path -Path $DownloadsFolder -ChildPath "Logins.Console.csproj";
+                if(Test-Path $ConsoleAppDestination)
+                {
+                    Remove-Item -Path $ConsoleAppDestination -Recurse;
+                }
+
+                # Remove version file
+                $VersionFileDestination = Join-Path -Path $DownloadsFolder -ChildPath "loginconsoleappversion.json";
+                if(Test-Path $VersionFileDestination)
+                {
+                    Remove-Item -Path $VersionFileDestination;
+                }
             }
 
-            #Testing Whether Console App is downloaded or not
-            $TestExePath =  Test-Path -Path $ExePath;
+            #Determine latest version of Login console app
+            $PackageId = "Microsoft.SqlServer.Migration.LoginsConsoleApp"
 
-            #Console app download address
-            $ZipSource = "https://sqlassess.blob.core.windows.net/app/LoginsMigration.zip";
-            $ZipDestination = Join-Path -Path $BaseFolder -ChildPath "LoginsMigration.zip";
+            $AvailablePackagesOnNugetOrg = ""
 
-            #Downloading and extracting LoginsMigration Zip file
-            if(-Not $TestExePath)
+            try {
+                $AvailablePackagesOnNugetOrg = Find-Package -Source "https://api.nuget.org/v3/index.json" -Name $PackageId -AllowPrereleaseVersions -AllVersions
```

Comment:
We should try to break it down into multiple functions. E.g. `CleanupOldVersion`, `GetLatestVersionNumberFromNuget`, `VerifyLocalVersion` , `DownloadNuget` for readability.

Reply:
Done